### ISSUE TITLE
Refactor `hda` to fix benchmark issues

### ISSUE THREAD

**Issue by tristantc** [SECQUOIA/gdplib#67]
### Description
This Issue aims to address and refactor the `hda` model to fix benchmark issues as outlined in [Issue #60](https://github.com/SECQUOIA/gdplib/issues/60).

Solution strategy causing issues:
- [x] Big-M MINLP reformulation
- [ ] Hull MINLP reformulation
- [ ] GLOA
- [x] Enumeration
- [x] LOA
- [x] LBB
- [x] RIC

### Branch
This PR targets the `benchmark_zd` branch.

### Steps to Reproduce
1. Run the benchmark tests for the `hda` model before applying the changes.
2. Apply the changes from this PR.
3. Re-run the benchmark tests to verify performance improvements.

### Notes
- Ensure that all sub-solvers are tested thoroughly after refactoring.
- Additional optimizations may be required based on benchmark results.

---

Feel free to adjust any details as necessary before submitting the PR.

**Comment by bernalde** [comment]
Benchmark evidence from PR #58 local/global Big-M smoke run (2026-05-10).

Configuration: `gdplib-benchmark run --strategies gdp.bigm --timelimit 300`; local profile = GAMS/DICOPT with IPOPTH/Gurobi role defaults; global profile = GAMS/BARON. Outputs are generated artifacts and are not committed.

| Profile | Solver | Termination | Status | Objective | Lower bound | Upper bound | User time (s) |
|---|---|---|---|---:|---:|---:|---:|
| local | DICOPT | optimal | ok | 5710.789875 | 5710.789875 | 6099.245522 | 300.065 |
| global | BARON | infeasible | ok |  |  | 1e+51 | 1.314 |

Warnings captured in build/Big-M transform pass:
- 354 unique warning event(s) in build/transform capture; categories: pyomo.core=354; examples: Setting Var 'a[1]' to a numeric value `1` outside the bounds (100, 10000). | Setting Var 'beta[1,ben]' to a numeric value `0.00011776` outside the bounds (0.00011776033005745731, 5.7264936018456645). | Setting Var 'conv[2,tol]' to a numeric value `2.12517` outside the bounds (0, 0.973).

Follow-up:
- Investigate global termination `infeasible` before treating the global bounds as final.
- Resolve the warning/deprecation items or link them to #55 if handled centrally.

Run artifacts: `benchmark_runs/pr58_bigm_local_5min_20260510`, `benchmark_runs/pr58_bigm_global_5min_20260510`, `benchmark_runs/pr58_bigm_warnings_transform_20260510`.

**Comment by bernalde** [comment]
**LOA row should be re-validated.** `hda` has a native maximization objective (`HDA_GDP_gdpopt.py:3316`, `sense=maximize`). The LOA checkmark above predates Pyomo/pyomo#3986 (merged 2026-08-13), which fixed GDPopt LOA hardcoding `minimize` sense in the OA master objective — for maximize models this could silently return wrong-direction "optimal" results (see #114 for the methanol evidence: 1 iteration, wrong disjuncts, still reported optimal).

Recommend re-running the LOA benchmark against a Pyomo build containing the fix and re-checking the recorded objective before considering the LOA row final. The fix is LOA-specific; GLOA/RIC/LBB results are unaffected.

**Comment by bernalde** [comment]
Coordination note from the PR #130 review cycle (evidence in the PR's review and address comments):

- The head model is feasible: GAMS/DICOPT (ipopth/gurobi roles) solves the Big-M reformulation to a local optimum at profit 1315.20.
- GAMS/BARON declares the same Big-M reformulation infeasible in presolve — a solver-side failure mirroring the BARON-infeasible pattern recorded for gdp_col in #66; both models received exp-form/eps1 domain-safety rewrites. Needs investigation before any global certificate is possible.
- The README's best-known 5801.27 is currently neither confirmable nor refutable; the PR body's GLOA "optimal" 1105.34 is a worse local solution than DICOPT's 1315.20, so `gams-local` GLOA labels are not global certificates.
- The HDA warm-start CSV is genuinely inconsistent with the model's Antoine-implied vapor-pressure bounds (`vp[*, h2]` initializers 0 vs lower bounds ≈64.5; 344 significant clamps logged at DEBUG by `_set_within_bounds`).

Remaining items this issue owns: reconcile the objective with a working global solver or enumeration, investigate the BARON presolve failure, update the README best-known value only after that, and audit the warm-start CSV against the model bounds.

**Comment by bernalde** [comment]
Correction and new evidence — supersedes the "BARON presolve pathology" framing in my earlier coordination note. A solver cross-check matrix (BARON, Gurobi, DICOPT × Big-M, hull; 120–300 s; `optcr=1e-6`, `threads=1`) shows the infeasibility is real and BARON is exonerated:

- Gurobi independently declares the hull-transformed head model infeasible in 0.6 s and finds no incumbent on Big-M in 120 s; BARON declares Big-M infeasible in 3.8 s and hull in 10.9 s.
- DICOPT's "optimal" points are not feasible: rechecking its returned point directly in Pyomo shows a reactor mass-balance (`Rctmbh2`) Big-M constraint violated by 1.8e-2 — far beyond tolerance. Every locally-solved hda result on record (including the earlier 1315.20 and the PR body's GLOA 1105.34) is therefore unreliable.
- The infeasibility pre-exists on main: Big-M + BARON declares main's hda infeasible in 1.7 s, so PR #130's rewrites did not introduce it.
- A Gurobi IIS on the hull model isolates 3 linear constraints, all in the `methane_flash_separation` disjunct's hull reformulation: the disaggregation identity for `t[44]`, the transformed `valp[4,58,4]` upper bound (disaggregated `p[49] <= p[47]`), and the disaggregated `eflsh[2,ch4] <= indicator` bound. Note `build_model` assigns per-index data-driven lb/ub to `eflsh` from CSV (around `HDA_GDP_gdpopt.py:1159`) — a likely place for a bound/data inconsistency.

Updated remaining items for this issue: (1) root-cause the `methane_flash_separation` infeasibility starting from the IIS above, (2) treat the README best-known 5801.27 as unverified for this GDP port until a feasible point exists, (3) drop the "investigate BARON" item — the solver was right.

**Comment by bernalde** [comment]
Update — the hda infeasibility is root-caused and fixed on PR #130 (commit 4b639b5; full chain in the PR comment). Summary for this tracker:

- Root cause: the eps1-in-log Antoine equations were exactly unsatisfiable against the Antoine-derived `vp` bounds (~1.3e-8 conflict, blocking residual 2.9e-3). Diagnosed by evaluating this port's constraints at the original gamslib `hda` model's optimal solution — the two models share stream numbering, which made point transfer possible.
- Fix: pole-free bilinear Antoine defining a bounded, shifted log-vapor-pressure auxiliary plus a bounded exponential; rate-constant prefactor 6.3e10 folded into the exponent. Model now verified feasible (CONOPT residuals ~1e-10; the point satisfies the full Big-M MINLP).
- Objective status: bounded 300 s global runs bracket the optimum in [5671.6, 5822.8] — the README's 5801.27 is plausible after all and stays, annotated as not yet certified. The earlier suspicion against it came from local-solver artifacts on the then-infeasible model, and gamslib's own 4322.55 is a DICOPT local claim.
- Remaining items this issue owns:
  1. Global certificate for the optimum (close the [5671.6, 5822.8] bracket; longer BARON/Gurobi runs).
  2. BARON false global certificates observed twice on this model family (cold "lb=ub=528.83" on an intermediate head; warm-started "lb=ub=4322.513" despite a verified 5671.62 feasible point) — worth a clean GAMS-native reproducer and, if it reproduces there, a report to the BARON developers.
  3. Hull + global solvers still report false infeasibility from the remaining `ratio`/compressor perspective numerics (huge FSG correction constants); the Antoine families are fixed, the same auxiliary treatment can be extended to the rest.
  4. The earlier note's warm-start CSV audit stands (`vp[*, h2]` initializers inconsistent with bounds).
